utils/sample_qc.py

```diff
@@ -0,0 +1,487 @@
+import numpy as np
+from utils.generic import *
+from gnomad_hail.resources import exome_calling_intervals_path
+from utils.gnomad_functions import logger, filter_to_adj
+import hdbscan
+
+
+def filter_rows_for_qc(
+        mt: hl.MatrixTable,
+        min_af: float = 0.001,
+        min_callrate: float = 0.99,
+        inbreeding_coeff_threshold: float = -0.8,
+        apply_hard_filters: bool = True
+) -> hl.MatrixTable:
+    """
+    Annotates rows with `sites_callrate`, `site_inbreeding_coeff` and `af`, then applies thresholds.
+    AF and callrate thresholds are taken from gnomAD QC, inbreeding coeff, MQ, FS and QD filter are taken from GATK best practices
+
+    :param MatrixTable mt: Input MT
+    :param float min_af: Minimum site AF to keep
+    :param float min_callrate: Minimum site call rate to keep
+    :param float inbreeding_coeff_threshold: Minimum site inbreeding coefficient to keep
+    :param bool apply_hard_filters: Whether to apply standard GAKT default site hard filters: QD >= 2, FS <= 60 and MQ >= 30
+    :return: annotated and filtered table
+    :rtype: MatrixTable
+    """
+    mt = mt.annotate_rows( # TODO: Make this an option if desired
+        site_callrate=hl.agg.fraction(hl.is_defined(mt.GT)),
+        site_inbreeding_coeff=bi_allelic_site_inbreeding_expr(mt.GT),
```

Comment:
So, I've rewritten things so that all of the filtering parameters are optional, with recommended defaults. I think it makes more sense.
Regarding inbreedingcoeff, it'll filter variants where there are way too many het samples given HWE. It's hard to think about a specific mode that would create these variants --  except maybe if you had a ton of related individuals from the same family then you wouldn't want this.
I'd use it for UKBB QC .

Reply:
Great, thank you! I will make sure to use this now for the UKBB QC!!